### PR TITLE
Fix #186, allow optional schema inference.

### LINKED ISSUE #186: TfDataValidator should be able to infer a schema when schema_path is none
In `write_stats_and_schema`, if schema is none, we can probably infer the schema from the stats file and upload it.

### PR DESCRIPTION
**Pull request by andrewsmartin**

### CODE REVIEW

**Comment by martin-laurent** [comment]
Great! Thanks!

File: spotify_tensorflow/tfx/tfdv.py [hunk 1]
```diff
@@ -71,7 +74,14 @@ def write_stats(self, pipeline_args):
...
 
     def write_stats_and_schema(self, pipeline_args):  # type: (List[str]) -> None
-        self.write_stats(pipeline_args)
+        stats = self.write_stats(pipeline_args)
+        if not self.schema:
+            logger.warning(
+                "Inferring a new schema for this dataset. If you want to use an existing schema, "
+                "provide a value for schema_path in the constructor."
+            )
+            new_schema = tfdv.infer_schema(stats)
```
Comment (daikeshi) [inline spotify_tensorflow/tfx/tfdv.py]: do we want to add `infer_feature_shape=false` to be more conservative? but i don't think it's a big deal

File: spotify_tensorflow/tfx/tfdv.py [hunk 2]
```diff
@@ -87,6 +97,11 @@ def validate_stats_against_schema(self):  # type: () -> bool
             return True
 
     def upload_schema(self):  # type: () -> None
+        if not self.schema:
+            raise Exception(
```
Comment (daikeshi) [inline spotify_tensorflow/tfx/tfdv.py]: minor, maybe use `ValueError`

Comment (andrewsmartin) [reply spotify_tensorflow/tfx/tfdv.py]: 👍

File: spotify_tensorflow/tfx/tfdv.py (hunk 1 above)
Comment (andrewsmartin) [reply spotify_tensorflow/tfx/tfdv.py]: Good catch, fixed!

**Pull request merged**